### ISSUE TITLE
wiki: URL encoding bug

### ISSUE THREAD

**Issue by ghost** [go-gitea/gitea#4090]
- Gitea version (or commit ref): 728cc27
- Git version: N/A
- Operating system: N/A
- Database (use `[x]`): N/A
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
    - https://try.gitea.io/yasuokav/link-test/wiki/Home
  - [ ] No
  - [ ] Not relevant
- Log gist: N/A

## Description
Since #4058, the links are no longer encoded.
But the URLs(files name) of the existing pages or new pages are still encoded.

The original link is
`https://try.gitea.io/yasuokav/link-test/wiki/Test%3F`
but now is
`https://try.gitea.io/yasuokav/link-test/wiki/Test?`

## Screenshots
N/A

**Comment by sapk** [comment]
I couldn't reproduce on the try.gitea.io link but maybe I didn't understand the bug.

**Comment by ghost** [comment]
@sapk When you click "Click Me" it will lead you to `https://try.gitea.io/yasuokav/link-test/wiki/_pages` 
instead of 
`https://try.gitea.io/yasuokav/link-test/wiki/Test%3F`

Because the URL is no longer encoded as `/yasuokav/link-test/wiki/Test%3F`
```
<a href="/yasuokav/link-test/wiki/Test?" rel="nofollow">Click Me</a>
```

**Comment by sapk** [comment]
I see, but that logical the file is name : Test%3F
I would see it more like a bug fixed by #4058
In your markdown, `[Click Me](Test?)` should be `[Click Me](Test%3F)`

If I take github as a reference : 
`[Click Me](Test?)` -> [Click Me](Test?)
`[Click Me](Test%3F)` -> [Click Me](Test%3F)

But we should notice it at the release that it could break some links.

**Comment by ghost** [comment]
@sapk Yes, you're right. But it will break some links if I use
```
[[Test?]]
```
Only `[[Test%3F]]` will work, but it looks bad.

Maybe we shouldn't encode the file name.

**Comment by sapk** [comment]
We need to encode the filename (in url) are those char are specials. But in case of [[Test?]] maybe url-encode the reference (link) to the page but not the title displayed.

**Comment by lunny** [comment]
should be closed by #4091
